Xwt/Xwt/MessageDialog.cs

```diff
@@ -32,8 +32,20 @@ namespace Xwt
 {
 	public static class MessageDialog
 	{
-		public static WindowFrame RootWindow { get; set; }
-		
+		private static WindowFrame rootWindow;
+
+		public static WindowFrame RootWindow
+		{
+			get {
+				if (Toolkit.CurrentEngine.Defaults.MessageDialog.DefaultRootWindowOverride != null)
+					return Toolkit.CurrentEngine.Defaults.MessageDialog.DefaultRootWindowOverride() ?? rootWindow;
```

Comment:
The API here is weird. Why do we want this as a `Func<WindowFrame>`?

Reply:
The problem is that if you show some dialog and a message using the message service, you need either to call an override with a custom parent window or set the RootWindow accordingly (and then back). It becomes even worse if you have different main windows (like VS for Mac has a main Shell and a GetToCode dialog). This also becomes a real problem if you consume some external API which is using Xwt and doesn't know your code. It won't know about your windows structure and always use RootWindow (like the Android SDK Manager in VS for Mac).